proposals/NNNN-localization.md

```diff
@@ -0,0 +1,107 @@
+
+## Proposed solution
+
+This proposal is about adding a localization class to the SDL library. This class can load the localization files bundled in the app based on a specified locale (language, region, script). The SDL manager can create localization objects based on the head unit language and provide those to the app developer.
+
+### Example for iOS:
+
+Today an app developer ready localized strings by using
+
+```objc
+NSLocalizedString(@"some-key"); // loading string of phone language
+```
+
+For SDL it would be
+
+```objc
+[self.manager.localization stringForKey:@"some-key"]; // loading string from SDL language
+```
+
+As of today the iOS library can automatically provide a localization instance using the SDL manager.
+
+### Example for Android:
+
+```java
+this.getResources().getText(R.string.some_key); // loading string from phone language
+```
+
+would be
+
+```java
+getLocalization().getText(R.string.some_key); // loading string from SDL language
+```
+
+The Android library don't contain SDL managers yet. The app developer can use the class manually if needed. Once Android includes a lifecycle manager it can automatically provide a localization object.
+
+### Detailed design
+
+First of all the localization code already exist:
+
+for Android see 
+[Localization.java](https://github.com/kshala-ford/sdl_android/blob/feature/localization/sdl_android_lib/src/com/smartdevicelink/util/Localization.java)
+
+and for iOS see 
+[SDLLocalization.h](https://github.com/kshala-ford/sdl_ios/blob/feature/localization/SmartDeviceLink-iOS/SmartDeviceLink/SDLLocalization.h) and [SDLLocalization.m](https://github.com/kshala-ford/sdl_ios/blob/feature/localization/SmartDeviceLink-iOS/SmartDeviceLink/SDLLocalization.m)
+
+#### Android
+
+For Android the Localization class is basically creating a new context with a custom locale and then provide the resource instance of the newly created context.
+
+```java
+Resources resources = context.getResources();
+Configuration config = new Configuration(resources.getConfiguration());
+config.setLocale(locale);
+ContextWrapper wrapper = new ContextWrapper(context);
+this.context = wrapper.createConfigurationContext(config);
+```
+
+This proposal is using an API called [createConfigurationContext()](https://developer.android.com/reference/android/content/ContextWrapper.html#createConfigurationContext(android.content.res.Configuration)) which was added at API level 17 (Android 4.2 Jelly Beans MR1). This method creates a new context separated from the apps configuration which allows the app to access resources of another language.
+
+The downside is the API level (see [Android dashboards](https://developer.android.com/about/dashboards/index.html)). As it is expected this number to become lower overtime and as it already reached a low number it is proposed to just use the phone language instead.
```

Comment:
Can you please change the last sentence to the following?

"As it is expected that this number will become lower over time, and as it has already reached a low number, it is proposed to just use the phone language instead."